logbook-spring-webflux/src/main/java/org/zalando/logbook/spring/webflux/LogbookExchangeFilterFunction.java

```diff
@@ -34,7 +35,12 @@ public Mono<org.springframework.web.reactive.function.client.ClientResponse> fil
                         .body((outputMessage, context) -> request.body().insert(new BufferingClientHttpRequest(outputMessage, clientRequest), context))
                         .build()
                 )
-                .flatMap(throwingFunction(response -> {
+                .doOnError(throwingConsumer(throwable -> {
+                            if (throwable.getCause() instanceof TimeoutException) {
```

Comment:
Not sure if it makes sense to capture only TimeoutException here, or try to log the request either way. 
On one hand, if the exchange failed prior to the sending, it may create a false assumption that the request happened. On the other hand, there can be other use cases, when the exchange failed after the request is sent, that are not related to timeouts.

Reply:
Plus, it looks like `io.netty.handler.timeout` package is not in the classpath, addressing TimeoutException would require adjustements to the dependency tree.

Reply:
I'm not super familiar with Netty; just asking out of curiosity. Looking at the `TimeoutException` inheritance tree:

```
java.lang.Object
  java.lang.Throwable
    java.lang.Exception
      java.lang.RuntimeException
        io.netty.channel.ChannelException
          io.netty.handler.timeout.TimeoutException
```

Would it make sense to generalize error handling to `ChannelException`?

Reply:
Seems like `io.netty.channel.ChannelException` is also not in the classpath. But I imagine it's not much farther away from catching any runtime exception at this point.

Reply:
Makes sense. What if we generalize it to `RuntimeException` then? In case the request is sent successfully, but any exception happens then (timeout, connection closed, etc.), we still want to log the request. This way, we won't need to change the classpath.

Reply:
So no condition for exceptions like here https://github.com/zalando/logbook/pull/2000/commits/77432d29014a57e8f6acd2252f58744df7845779 ?

Reply:
Yes, I think it's the easiest way to fix it 👍